packages/solidity-contracts/scripts/hardhat/verifyDeployment.ts

```diff
@@ -0,0 +1,95 @@
+import { task } from 'hardhat/config';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { config as dotEnvConfig } from 'dotenv';
+
+dotEnvConfig();
+
+task('verify-deployment', 'Verifies the deployed contract bytecode').setAction(
+  async (taskArgs: any, hre: HardhatRuntimeEnvironment): Promise<void> => {
+    const network = hre.network.name;
+    console.log(
+      `Verifying contract bytecode on ${network}:${hre.network.config.chainId}...`
+    );
+
+    const deployments = await hre.deployments.all();
+
+    for (const [contractName, deployment] of Object.entries(deployments)) {
+      console.log(`\nVerifying ${contractName} (${deployment.address}):`);
+
+      console.log('--- Fetching deployed bytecode...');
+
+      if (!deployment.linkedData.isProxy) {
+        console.log('--- Creating local Hardhat network...');
+        const localHardhat = require('hardhat');
+        await localHardhat.run('compile');
+
+        const ContractFactory = await localHardhat.ethers.getContractFactory(
+          deployment.linkedData.factory
+        );
+
+        console.log('--- Validating Upgrade...');
```

Comment:
I think that either here or somewhere else where it makes sense, we should add a reference to the new implementation address that we are checking